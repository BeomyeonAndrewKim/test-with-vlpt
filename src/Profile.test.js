import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

const wrapper = mount(<Profile username="Andrea" name="beomyeon" />);
describe("<Profile />", () => {
  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders username and name", () => {
    expect(wrapper.props().username).toBe("Andrea");
    expect(wrapper.props().name).toBe("beomyeon");

    const boldElement = wrapper.find("b");
    expect(boldElement.contains("Andrea")).toBe(true);
    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("(beomyeon)");
  });
});
