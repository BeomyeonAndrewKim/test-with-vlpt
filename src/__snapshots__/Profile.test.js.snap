// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="beomyeon"
  username="Andrea"
>
  <div>
    <b>
      Andrea
      !
    </b>
    Â 
    <span>
      (
      beomyeon
      )
    </span>
  </div>
</Profile>
`;
